Specifications/Language/3_Expressions/ConditionalLoops.md

```diff
@@ -36,21 +34,21 @@ until condition;
 
 where `condition` is an arbitrary expression of type `Bool`.
 
-The `repeat` statement runs a block of statements before evaluating a condition. If the condition evaluates to true, the loop exits. If the condition evaluates to false, an additional block of statements defined as part of an optional `fixup` block, if present, is run prior to entering the next loop iteration. 
+The `repeat` loop runs a block of statements before evaluating a condition. If the condition evaluates to true, the loop exits. If the condition evaluates to false, an additional block of statements defined as part of an optional `fixup` block, if present, is run prior to entering the next loop iteration.
 
-The compiler treats all parts of the `repeat` statement (both blocks and the condition) as a single scope for each repetition; symbols that are defined within the `repeat` block are visible both to the condition and within the `fixup` block. As for other loops, symbols go out of scope after each iteration, such that symbols defined in the `fixup` block are not visible in the `repeat` block.
+The compiler treats all parts of the `repeat` loop (both blocks and the condition) as a single scope for each repetition; symbols that are defined within the `repeat` block are visible both to the condition and within the `fixup` block. As for other loops, symbols go out of scope after each iteration, such that symbols defined in the `fixup` block are not visible in the `repeat` block.
 
 ### Target-specific restrictions
 
-Loops that break based on a condition are challenging to process on quantum hardware if the condition depends on measurement outcomes since the length of the instruction sequence to run is not known ahead of time. 
+Loops that break based on a condition are challenging to process on quantum hardware if the condition depends on measurement outcomes since the length of the instruction sequence to run is not known ahead of time.
 
 Despite their common presence in particular classes of quantum algorithms, current hardware does not yet provide native support for these kinds of control flow constructs. Running on quantum hardware can potentially be supported in the future by imposing a maximum number of iterations, or as additional hardware support becomes available.
 
 ## While loop
 
-A more familiar-looking statement for classical computations is the `while` loop. It is supported only within functions. 
+A more familiar-looking loop for classical computations is the `while` loop. It is supported only within functions.
```

Comment:
This is also no longer true, while loops can now appear in operations. This makes explaining things easier, as operations are a strict superset of functions, and the compiler checks only need to verify that functions don't use any features they aren't allowed to use.

Reply:
Is it just that functions don't allow repeat-until loops?

Reply:
Yup. Functions can't use the following (as seen in [the pass that checks for these errors](https://github.com/microsoft/qsharp/blob/main/compiler/qsc_passes/src/callable_limits.rs)):
 - conjugate expressions
 - functor declarations
 - specialization declarations
 - calls to operations
 - qubit allocation statements
 - repeat-until loop expressions

Operations no longer have any limitations.

Reply:
updated to remove the sentnece about the restriction.